### PR TITLE
feat: model registry tab in workspace

### PR DESCRIPTION
**Pull request by keita-determined**
## Description

[WEB-803]

## Test Plan
- If RBAC is NOT enabled
  - Everyone can see model registry tab in workspace and global model registry
- If RBAC is enabled
  - User who has `VIEWMODELREGISTRY` permission can see model registry tab in workspace and global model registry
- For Both (RBAC is enabled and disabled)
  - Check if `useSettings` are independent
    - Manually change the workspaceId col in db to check since we dont have workspace dropdown in model modal.

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

[WEB-803]: https://determinedai.atlassian.net/browse/WEB-803?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/hooks/usePermissions.ts [hunk 1]
```diff
@@ -230,6 +232,20 @@ const canViewGroups = ({ rbacReadPermission, rbacEnabled, user }: RbacOptsProps)
...
+  rbacReadPermission,
+  rbacEnabled,
+  user,
+  userAssignments,
+  userRoles,
+}: RbacOptsProps): boolean => {
+  const permitted = relevantPermissions(userAssignments, userRoles);
+  return (
+    rbacReadPermission ||
+    (!!user && (rbacEnabled ? permitted.has(V1PermissionType.VIEWMODELREGISTRY) : user.isAdmin))
```
Comment (keita-determined) [inline webui/react/src/hooks/usePermissions.ts:245]: Is it sufficient?

Comment (julian-determined-ai) [reply webui/react/src/hooks/usePermissions.ts:245]: @keita-determined as written isd `rbac` is not enabled, then only `administrators` can view the model registry. Is that what we want?

Comment (keita-determined) [reply webui/react/src/hooks/usePermissions.ts:245]: Thats a good question. I had the same question. Let me ask in slack channel

Comment (keita-determined) [reply webui/react/src/hooks/usePermissions.ts]: @julian-determined-ai 
Everyone can see model registry tab in workspace and global model registry for OSS
changed the code

File: webui/react/src/hooks/usePermissions.ts [hunk 2]
```diff
@@ -230,6 +232,20 @@ const canViewGroups = ({ rbacReadPermission, rbacEnabled, user }: RbacOptsProps)
...
+  rbacReadPermission,
+  rbacEnabled,
+  user,
+  userAssignments,
+  userRoles,
+}: RbacOptsProps): boolean => {
+  const permitted = relevantPermissions(userAssignments, userRoles);
+  return (
+    rbacReadPermission ||
+    (!!user && (rbacEnabled ? permitted.has(V1PermissionType.VIEWMODELREGISTRY) : true))
```
Comment (julian-determined-ai) [inline webui/react/src/hooks/usePermissions.ts:245]: @keita-determined we can simplify this more. How about something such as 

```
rbacReadPermission || (!!user && (!rbacEnabled || permitted.has(V1PermissionType.VIEWMODELREGISTRY)))
```

Comment (keita-determined) [reply webui/react/src/hooks/usePermissions.ts]: ok,  how about this? i also put a comment 

```ts
!rbacEnabled || rbacReadPermission || permitted.has(V1PermissionType.VIEWMODELREGISTRY)
```

**Review by julian-determined-ai — APPROVED** [review]
Looks great!

**Pull request merged**